src/platform/i386/exception.c

```diff
@@ -9,7 +9,8 @@
 void
 print_regs_state(struct pt_regs *regs)
 {
-	printk("\n\nGeneral registers-> EAX: %x, EBX: %x, ECX: %x, EDX: %x\n", regs->ax, regs->bx, regs->cx, regs->dx);
+	printk("\n\nCPU%d registers:\n", get_cpuid());
```

Comment:
This is not very useful! I think we need `get_cpuid()` in every printk below. 
You may have noticed that the interleavings among cores just makes it impossible to make any sense of the output.. To be able to understand anything that's going on (mainly if fewer cores fault), then we need to know which of the below statements (registers) correspond to which core..

Reply:
This is really based on the experience in debugging some of the issues. I totally understand why you've this here. (Qemu just cannot interleave like the real hw..)..